react/features/screenshot-capture/actions.js

```diff
@@ -28,19 +30,25 @@ function setScreenshotCapture(enabled) {
 * @returns {Promise}
 */
 export function toggleScreenshotCaptureEffect(enabled: boolean) {
-    return function(dispatch: (Object) => Object, getState: () => any) {
+    return async function(dispatch: (Object) => Object, getState: () => any) {
         const state = getState();
 
         if (state['features/screenshot-capture'].capturesEnabled !== enabled) {
             const { jitsiTrack } = getLocalVideoTrack(state['features/base/tracks']);
 
+            if (!ongoingEffect) {
+                ongoingEffect = await createScreenshotCaptureEffect(state);
+            }
+
             // Screenshot capture effect doesn't return a modified stream. Therefore, we don't have to
             // switch the stream at the conference level, starting/stopping the effect will suffice here.
-            return createScreenshotCaptureEffect(state)
-                .then(effect => {
-                    enabled ? effect.startEffect(jitsiTrack.getOriginalStream()) : effect.stopEffect();
-                    dispatch(setScreenshotCapture(enabled));
-                });
+            enabled
+                ? ongoingEffect.startEffect(
+                    jitsiTrack.getOriginalStream(),
+                    jitsiTrack.videoType
+                )
+                : ongoingEffect.stopEffect();
+            dispatch(setScreenshotCapture(enabled));
```

Comment:
Can we make sure we dispatch setScreenshotCapture only when Promise resolves for startEffect ?